app/src/components/common/SharingModal/ShareLinkView.tsx

```diff
@@ -0,0 +1,246 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import { useSelector } from "react-redux";
+import { getAllRules, getAppMode, getGroupwiseRulesToPopulate, getUserAuthDetails } from "store/selectors";
+import { Radio, Space } from "antd";
+import { RQButton, RQInput } from "lib/design-system/components";
+import { CopyValue } from "components/misc/CopyValue";
+import { createSharedList } from "components/features/sharedLists/CreateSharedListModal/actions";
+import { ReactMultiEmail, isEmail as validateEmail } from "react-multi-email";
+import { epochToDateAndTimeString } from "utils/DateTimeUtils";
+import { getSharedListURL } from "utils/PathUtils";
+import { toast } from "utils/Toast";
+import { httpsCallable, getFunctions } from "firebase/functions";
+import { CheckCircleFilled } from "@ant-design/icons";
+import { SharedLinkVisibility } from "./types";
+import { Rule } from "types";
+import Logger from "lib/logger";
+import "./index.css";
+
+interface ShareLinkProps {
+  rulesToShare: string[];
+}
+
+// TODO: handle copy changes for session replay in V1
+
+export const ShareLinkView: React.FC<ShareLinkProps> = ({ rulesToShare }) => {
+  const user = useSelector(getUserAuthDetails);
+  const appMode = useSelector(getAppMode);
+  const rules = useSelector(getAllRules);
+  const groupwiseRulesToPopulate = useSelector(getGroupwiseRulesToPopulate);
+  const [sharedLinkVisibility, setSharedLinkVisibility] = useState(SharedLinkVisibility.PUBLIC);
+  const [permittedEmailsList, setPermittedEmailsList] = useState([]);
+  const [sharedListRecipients, setSharedListRecipients] = useState([]);
+  const [sharedListName, setSharedListName] = useState(null);
+  const [shareableLinkData, setShareableLinkData] = useState(null);
+  const [isLinkGenerating, setIsLinkGenerating] = useState(false);
+  const [isMailSent, setIsMailSent] = useState(false);
+  const [error, setError] = useState(null);
+
+  const sendSharedListShareEmail = useMemo(() => httpsCallable(getFunctions(), "sharedLists-sendShareEmail"), []);
+  const singleRuleData = useMemo(
+    () => (rulesToShare && rulesToShare?.length === 1 ? rules.find((rule: Rule) => rule.id === rulesToShare[0]) : null),
+    [rules, rulesToShare]
+  );
+
+  const visibilityOptions = useMemo(
+    () => [
+      {
+        label: "Public shared list",
+        value: SharedLinkVisibility.PUBLIC,
+        description: "Anyone with a link can access the rule.",
+      },
+      {
+        label: "Private shared list",
+        value: SharedLinkVisibility.PRIVATE,
+        description: "Only accessible only by accounts you specify.",
+      },
+    ],
+    []
+  );
+
+  const renderSharedListNameField = () => {
+    return (
```

Comment:
no need to make this as a function directly assign `sharedListNameField = <>....</>`